tezos/context/src/working_tree/storage.rs

```diff
@@ -616,14 +622,17 @@ impl Storage {
         )
     }
 
-    fn binary_search_in_tree(
+    /// Search `key` in the directory and return the result of `slice::binary_search_by`.
+    ///
+    /// `dir` must be sorted.
+    fn binary_search_in_dir(
         &self,
-        tree: &[(StringId, NodeId)],
+        dir: &[(StringId, NodeId)],
         key: &str,
     ) -> Result<Result<usize, usize>, StorageError> {
```

Comment:
This result type is very weird, please add some more text in the comment explaining it.